src/goto-instrument/contracts/utils.cpp

```diff
@@ -154,6 +154,19 @@ void insert_before_swap_and_advance(
   std::advance(target, offset);
 }
 
+void insert_before_and_update_jumps(
+  goto_programt &destination,
+  goto_programt::targett &target,
+  const goto_programt::instructiont &i)
+{
+  const auto new_target = destination.insert_before(target, i);
+  for(auto it : target->incoming_edges)
+  {
+    if(it->is_goto())
```

Comment:
should we error out if happen if another type of instructions than GOTO is found in `incoming_edges` ?

Reply:
No, ```incoming_edges``` includes GOTO jumps, and the instruction iterator right before ```target```. So it common that there is one incoming edge from a non-GOTO instruction.